src/NativeScript/inlineFunctions.js

```diff
@@ -14,6 +14,119 @@ Object.assign(global, {
+                if (d = decorators[i]) {
+                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
+                }
+            }
+        }
+        return c > 3 && r && Object.defineProperty(target, key, r), r;
+    },
+    __param(paramIndex, decorator) {
+        return function (target, key) {
+            decorator(target, key, paramIndex);
+        }
+    },  
+
+    // Decorators
+    ObjCProtocols() {
+        var protocols = [];
+        for (var i = 0; i < arguments.length; i++) {
+            protocols[i] = arguments[i];
+        }
+
+        return function (target) {
+            target.ObjCProtocols = protocols.length > 0 ? protocols : undefined;
+        };
+    },
+    ObjCMethod(name, returnType) {
+        return function (target, propertyKey, descriptor) {
+            if (!target.constructor.ObjCExposedMethods) {
+                target.constructor.ObjCExposedMethods = {};
+            }
+            if (!target.constructor.ObjCExposedMethods[propertyKey]) {
+                target.constructor.ObjCExposedMethods[propertyKey] = {};
+            }
+            target.constructor.ObjCExposedMethods[propertyKey].returns = returnType || interop.types.void;
+
+            if (name !== propertyKey) {
+                target.constructor.ObjCExposedMethods[name] = target.constructor.ObjCExposedMethods[propertyKey];
+                delete target.constructor.ObjCExposedMethods[propertyKey];
+
+                target[name] = function () { 
+                    return this[propertyKey].apply(this, arguments);
+                }
+            }    
+        };
+    },    
+    ObjCParam(type) {
+        return function (target, propertyKey, parameterIndex) {
+            if (!target.constructor.ObjCExposedMethods) {
+                target.constructor.ObjCExposedMethods = {};
+            }
+            if (!target.constructor.ObjCExposedMethods[propertyKey]) {
+                target.constructor.ObjCExposedMethods[propertyKey] = {};
+            }
+            var exposedMethod = target.constructor.ObjCExposedMethods[propertyKey];
+            if (!exposedMethod.params) {
+                exposedMethod.params = [];
+            }
+            exposedMethod.params[parameterIndex] = type || interop.types.void;
+        };
+    },
+    // ObjC(name) {
+    //     var additionalParams = [];
+    //     for (var i = 1; i < arguments.length; i++) {
+    //         additionalParams[i - 1] = arguments[i];
+    //     }
+
+    //     return function (target, propertyKey, descriptor) {
+    //         if (propertyKey === undefined) { // Class decorator
+    //             var implementation = {};
+    //             for (var key in target.prototype) {
+    //                 if (target.prototype.hasOwnProperty(key)) {
+    //                     implementation[key] = target[key];
+    //                 }    
+    //             }
+    //             var extended = target.extend(implementation, {
```

Comment:
@ivanbuhov noted the `Object.getPrototypeOf(target)` will work too. The modern JavaScript and the runtime wires the prototype chain of the constructor functions properly, not only the prototype chain of the prototypes objects. This is so cool. :+1: